src/thor/triplegbuilder.cc

```diff
@@ -1376,7 +1382,7 @@ void TripLegBuilder::Build(
   }
 
   // Add that extra costing information if requested
-  AccumulateRecostingInfoForward(options, start_pct, end_pct, date_time, graphreader, trip_path);
+  AccumulateRecostingInfoForward(options, start_pct, end_pct, time_info, graphreader, trip_path);
```

Comment:
i see you want to use the unchanged one here. if that is the case then i suggest you make a copy of it to modify in the loop to avoid unneccesary timezone offsets and look ups